acra/acra-in-depth/data-flow/_index.md

```diff
@@ -1,4 +1,113 @@
 ---
-title: Data flow(s)
+title: Data flow
 bookCollapseSection: true
 ---
+
+## Data flow
+
+### Acra components
+
+- [_AcraWriter_](/pages/documentation-acra/#acraconnector-and-acrawriter) — a client-side library that integrates into the app's workflow either through ORM or directly and provides the means for encryption of the sensitive data by generating AcraStructs.
+
+- [_AcraServer_]({{< ref "/acra/configuring-maintaining/general-configuration/acra_server.md" >}}) — a separate daemon that runs in an isolated environment (separate virtual machine or physical server). AcraServer is responsible for holding all the secrets required to decrypt the data and for actually decrypting this data.
+
+- [_AcraCensor_]({{< ref "/acra/security-controls/sql-firewall/" >}}) – is a firewall-like component that sits inside AcraServer, and checks every SQL query to the database.
+
+  ![](/files/data-flow/acra-archi-tls.png)
+  
+  _Acra-based application architecture using TLS transport encryption._
+  
+Second way to protect raw data between application and AcraServer/AcraTranslator is to use AcraConnector:
+
+- [_AcraConnector_]({{< ref "/acra/configuring-maintaining/general-configuration/acra_connector.md" >}}) — a client-side daemon that runs under a separate user / in a separate container, and which acts as a database listener that redirects all the queries to AcraServer and, upon receiving the results, feeds it back into the app. AcraConnector is an optional component, required in systems that are using extra transport encryption layer via Themis SecureSession.
+
+  ![](/files/data-flow/acra-entities.png)    
+  
+  _Acra-based application architecture using TLS transport encryption._
+  
+### Data flow schemas
+
+With TLS as secure transport between application and AcraServer/AcraTranslator:
+1. **Writing**: Application --> AcraTranslator --> Application --> Storage.
+
+   **Description**: application uses AcraTranslator to encrypt/decrypt data and save it in a storage. 
+  
+   **Reading**: Application --> Storage --> Application --> AcraTranslator --> Application.
+
+   **Description**: application uses AcraTranslator to encrypt/decrypt data and save it in a storage.
```

Comment:
Copy-paste of a description above (line 33).
Maybe make it `application uses AcraTranslator to decrypt data after reading from storage`?
And in line 33 make it `encrypt data` only?

Similar thing is in lines 73 & 78, maybe it was intended...

Reply:
nice catch. yes, probably copy-paste. applied your suggestion and updated descriptions